2025-09-05 19:44:05 [INFO] base.BaseTest - =========================================
2025-09-05 19:44:05 [INFO] base.BaseTest - ======== Test Execution Started =========
2025-09-05 19:44:05 [INFO] base.BaseTest - =========================================
2025-09-05 19:44:05 [INFO] base.BaseTest - Selected Browser: chrome
2025-09-05 19:44:18 [DEBUG] org.apache.poi.openxml4j.opc.PackageRelationshipCollection - Parsing relationship: /xl/_rels/workbook.xml.rels
2025-09-05 19:44:18 [DEBUG] org.apache.poi.openxml4j.opc.PackageRelationshipCollection - Parsing relationship: /_rels/.rels
2025-09-05 19:44:21 [INFO] utils.TestListener - ===== STARTING TEST: loginTest =====
2025-09-05 19:44:21 [INFO] base.BaseTest - ------------------------------------------------------------
2025-09-05 19:44:21 [INFO] base.BaseTest - ▶ Test Case: 1
2025-09-05 19:44:21 [INFO] base.BaseTest -    ➤ Desc     : Valid login with standard user
2025-09-05 19:44:21 [INFO] base.BaseTest -    ➤ Type     : Positive
2025-09-05 19:44:21 [INFO] base.BaseTest -    ➤ Groups   : smoke, regression,sanity
2025-09-05 19:44:21 [INFO] base.BaseTest -    ➤ Username : standard_user
2025-09-05 19:44:21 [INFO] base.BaseTest -    ➤ Expected : success
2025-09-05 19:44:21 [INFO] base.BaseTest - ------------------------------------------------------------
2025-09-05 19:44:22 [INFO] pages.LoginPage - Entered 'standard_user' into: UserName Field
2025-09-05 19:44:23 [INFO] pages.LoginPage - Entered 'secret_sauce' into: PasswordField
2025-09-05 19:44:23 [INFO] pages.LoginPage - Clicked on: Login Button
2025-09-05 19:44:29 [INFO] pages.LoginPage - Login submitted, navigating to Home Page.
2025-09-05 19:44:32 [INFO] base.BaseTest - 1 PASSED ✅ : User logged in successfully.
2025-09-05 19:44:33 [INFO] utils.TestListener - ✅ TEST PASSED: loginTest
2025-09-05 19:44:33 [INFO] utils.TestListener - ===== STARTING TEST: loginTest =====
2025-09-05 19:44:33 [INFO] base.BaseTest - ------------------------------------------------------------
2025-09-05 19:44:33 [INFO] base.BaseTest - ▶ Test Case: 2
2025-09-05 19:44:33 [INFO] base.BaseTest -    ➤ Desc     : Invalid login with wrong password
2025-09-05 19:44:33 [INFO] base.BaseTest -    ➤ Type     : Negative
2025-09-05 19:44:33 [INFO] base.BaseTest -    ➤ Groups   : regression
2025-09-05 19:44:33 [INFO] base.BaseTest -    ➤ Username : standard_user
2025-09-05 19:44:33 [INFO] base.BaseTest -    ➤ Expected : failure
2025-09-05 19:44:33 [INFO] base.BaseTest - ------------------------------------------------------------
2025-09-05 19:44:39 [INFO] pages.LoginPage - Entered 'standard_user' into: UserName Field
2025-09-05 19:44:43 [INFO] pages.LoginPage - Entered 'wrong_pass' into: PasswordField
2025-09-05 19:44:45 [INFO] pages.LoginPage - Clicked on: Login Button
2025-09-05 19:44:47 [ERROR] pages.LoginPage - Login failed with error: Epic sadface: Username and password do not match any user in this service
2025-09-05 19:44:47 [ERROR] utils.TestListener - ❌ TEST FAILED: loginTest
java.lang.AssertionError: Login failed with error: Epic sadface: Username and password do not match any user in this service
	at org.junit.Assert.fail(Assert.java:89) ~[junit-4.13.2.jar:4.13.2]
	at pages.LoginPage.clickLoginButton(LoginPage.java:52) ~[test-classes/:?]
	at tests.DemoPage.loginTest(DemoPage.java:47) ~[test-classes/:?]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:686) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:992) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134) ~[testng-7.10.1.jar:7.10.1]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596) [?:?]
	at org.testng.TestRunner.privateRun(TestRunner.java:739) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestRunner.run(TestRunner.java:614) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:421) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.run(SuiteRunner.java:312) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1208) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.runSuites(TestNG.java:1112) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.run(TestNG.java:1079) [testng-7.10.1.jar:7.10.1]
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115) [testng-remote.jar:?]
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293) [testng-remote.jar:?]
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91) [testng-remote.jar:?]
2025-09-05 19:44:47 [ERROR] base.BaseTest - ❌ Test Failed: loginTest
2025-09-05 19:44:48 [INFO] utils.TestListener - ===== STARTING TEST: loginTest =====
2025-09-05 19:44:48 [INFO] base.BaseTest - ------------------------------------------------------------
2025-09-05 19:44:48 [INFO] base.BaseTest - ▶ Test Case: 3
2025-09-05 19:44:48 [INFO] base.BaseTest -    ➤ Desc     : Login attempt for locked-out user
2025-09-05 19:44:48 [INFO] base.BaseTest -    ➤ Type     : Negative
2025-09-05 19:44:48 [INFO] base.BaseTest -    ➤ Groups   : regression
2025-09-05 19:44:48 [INFO] base.BaseTest -    ➤ Username : locked_out_user
2025-09-05 19:44:48 [INFO] base.BaseTest -    ➤ Expected : failure
2025-09-05 19:44:48 [INFO] base.BaseTest - ------------------------------------------------------------
2025-09-05 19:44:48 [INFO] pages.LoginPage - Entered 'locked_out_user' into: UserName Field
2025-09-05 19:44:49 [INFO] pages.LoginPage - Entered 'secret_sauce' into: PasswordField
2025-09-05 19:44:49 [INFO] pages.LoginPage - Clicked on: Login Button
2025-09-05 19:44:51 [ERROR] pages.LoginPage - Login failed with error: Epic sadface: Sorry, this user has been locked out.
2025-09-05 19:44:51 [ERROR] utils.TestListener - ❌ TEST FAILED: loginTest
java.lang.AssertionError: Login failed with error: Epic sadface: Sorry, this user has been locked out.
	at org.junit.Assert.fail(Assert.java:89) ~[junit-4.13.2.jar:4.13.2]
	at pages.LoginPage.clickLoginButton(LoginPage.java:52) ~[test-classes/:?]
	at tests.DemoPage.loginTest(DemoPage.java:47) ~[test-classes/:?]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:686) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:992) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134) ~[testng-7.10.1.jar:7.10.1]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596) [?:?]
	at org.testng.TestRunner.privateRun(TestRunner.java:739) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestRunner.run(TestRunner.java:614) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:421) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.run(SuiteRunner.java:312) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1208) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.runSuites(TestNG.java:1112) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.run(TestNG.java:1079) [testng-7.10.1.jar:7.10.1]
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115) [testng-remote.jar:?]
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293) [testng-remote.jar:?]
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91) [testng-remote.jar:?]
2025-09-05 19:44:51 [ERROR] base.BaseTest - ❌ Test Failed: loginTest
2025-09-05 19:44:52 [INFO] utils.TestListener - ===== STARTING TEST: loginTest =====
2025-09-05 19:44:52 [INFO] base.BaseTest - ------------------------------------------------------------
2025-09-05 19:44:52 [INFO] base.BaseTest - ▶ Test Case: 4
2025-09-05 19:44:52 [INFO] base.BaseTest -    ➤ Desc     : Blank username and password fields
2025-09-05 19:44:52 [INFO] base.BaseTest -    ➤ Type     : Edge
2025-09-05 19:44:52 [INFO] base.BaseTest -    ➤ Groups   : regression,sanity
2025-09-05 19:44:52 [INFO] base.BaseTest -    ➤ Username : ''
2025-09-05 19:44:52 [INFO] base.BaseTest -    ➤ Expected : failure
2025-09-05 19:44:52 [INFO] base.BaseTest - ------------------------------------------------------------
2025-09-05 19:44:52 [INFO] pages.LoginPage - Entered '''' into: UserName Field
2025-09-05 19:44:53 [INFO] pages.LoginPage - Entered '''' into: PasswordField
2025-09-05 19:44:54 [INFO] pages.LoginPage - Clicked on: Login Button
2025-09-05 19:44:56 [ERROR] pages.LoginPage - Login failed with error: Epic sadface: Username and password do not match any user in this service
2025-09-05 19:44:56 [ERROR] utils.TestListener - ❌ TEST FAILED: loginTest
java.lang.AssertionError: Login failed with error: Epic sadface: Username and password do not match any user in this service
	at org.junit.Assert.fail(Assert.java:89) ~[junit-4.13.2.jar:4.13.2]
	at pages.LoginPage.clickLoginButton(LoginPage.java:52) ~[test-classes/:?]
	at tests.DemoPage.loginTest(DemoPage.java:47) ~[test-classes/:?]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:686) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:992) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134) ~[testng-7.10.1.jar:7.10.1]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596) [?:?]
	at org.testng.TestRunner.privateRun(TestRunner.java:739) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestRunner.run(TestRunner.java:614) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:421) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.run(SuiteRunner.java:312) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1208) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.runSuites(TestNG.java:1112) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.run(TestNG.java:1079) [testng-7.10.1.jar:7.10.1]
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115) [testng-remote.jar:?]
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293) [testng-remote.jar:?]
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91) [testng-remote.jar:?]
2025-09-05 19:44:56 [ERROR] base.BaseTest - ❌ Test Failed: loginTest
2025-09-05 19:44:56 [INFO] utils.TestListener - ===== STARTING TEST: loginTest =====
2025-09-05 19:44:56 [INFO] base.BaseTest - ------------------------------------------------------------
2025-09-05 19:44:56 [INFO] base.BaseTest - ▶ Test Case: 5
2025-09-05 19:44:56 [INFO] base.BaseTest -    ➤ Desc     : SQL injection attempt
2025-09-05 19:44:56 [INFO] base.BaseTest -    ➤ Type     : Edge
2025-09-05 19:44:56 [INFO] base.BaseTest -    ➤ Groups   : regression
2025-09-05 19:44:56 [INFO] base.BaseTest -    ➤ Username : ' OR '1'='1
2025-09-05 19:44:56 [INFO] base.BaseTest -    ➤ Expected : failure
2025-09-05 19:44:56 [INFO] base.BaseTest - ------------------------------------------------------------
2025-09-05 19:44:57 [INFO] pages.LoginPage - Entered '' OR '1'='1' into: UserName Field
2025-09-05 19:44:57 [INFO] pages.LoginPage - Entered 'secret_sauce' into: PasswordField
2025-09-05 19:44:58 [INFO] pages.LoginPage - Clicked on: Login Button
2025-09-05 19:45:00 [ERROR] pages.LoginPage - Login failed with error: Epic sadface: Username and password do not match any user in this service
2025-09-05 19:45:00 [ERROR] utils.TestListener - ❌ TEST FAILED: loginTest
java.lang.AssertionError: Login failed with error: Epic sadface: Username and password do not match any user in this service
	at org.junit.Assert.fail(Assert.java:89) ~[junit-4.13.2.jar:4.13.2]
	at pages.LoginPage.clickLoginButton(LoginPage.java:52) ~[test-classes/:?]
	at tests.DemoPage.loginTest(DemoPage.java:47) ~[test-classes/:?]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:686) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:992) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154) ~[testng-7.10.1.jar:7.10.1]
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134) ~[testng-7.10.1.jar:7.10.1]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596) [?:?]
	at org.testng.TestRunner.privateRun(TestRunner.java:739) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestRunner.run(TestRunner.java:614) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:421) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunner.run(SuiteRunner.java:312) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52) [testng-7.10.1.jar:7.10.1]
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1208) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.runSuites(TestNG.java:1112) [testng-7.10.1.jar:7.10.1]
	at org.testng.TestNG.run(TestNG.java:1079) [testng-7.10.1.jar:7.10.1]
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115) [testng-remote.jar:?]
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293) [testng-remote.jar:?]
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91) [testng-remote.jar:?]
2025-09-05 19:45:00 [ERROR] base.BaseTest - ❌ Test Failed: loginTest
2025-09-05 19:45:02 [INFO] base.BaseTest - Browser closed.
2025-09-05 19:45:02 [INFO] base.BaseTest - ===== Test Execution Finished =====
2025-09-05 19:45:02 [INFO] utils.TestListener - ===== FINISHED TEST SUITE: Confirmation Tests =====
